src/serac/numerics/functional/quadrature_data.hpp

```diff
@@ -54,54 +107,64 @@ struct Empty {
  */
 template <typename T>
 struct QuadratureData {
-  /// ctor, allocates memory and sets up strides
-  QuadratureData(size_t n1, size_t n2) : stride(n2) { data = new T[n1 * n2]; }
-
-  /// dtor, deallocates memory
-  ~QuadratureData() { delete data; }
 
-  /// access a mutable reference to the quadrature data at element `i`, quadrature point `j`
-  SERAC_HOST_DEVICE T& operator()(size_t i, size_t j) { return data[i * stride + j]; }
+  using geom_array_t = std::array< uint32_t, mfem::Geometry::NUM_GEOMETRIES >;
 
-  /// access a const reference to the quadrature data at element `i`, quadrature point `j`
-  SERAC_HOST_DEVICE const T& operator()(size_t i, size_t j) const { return data[i * stride + j]; }
+  using tmp_t = axom::Array<T, 2>;
```

Comment:
I'll go back and rename the things in this class that I forgot to do originally. 

---------

To answer your question so you don't have to go reread the comments after I write them:

Before, meshes were only allowed to have one element geometry, so when we allocated memory for quadrature point data it was just a 2D array, with indices describing:
(which element, which qpt within that element).

Now, meshes are allowed to have different kinds of element geometries. So, when we allocate space for that quadrature point data, it's a 3D array indexed by:
(which geometry, which element with that geometry, which qpt within that element)

This 3D array is implemented as a `std::map` of 2D axom arrays, since the leftmost index is "sparse" (only a couple of the geometry types will be used for a given mesh). `tmp_t` was a (bad, temporary) name given to the 2D array before I changed it to use `axom::Array`.